### PR TITLE
Bogwar/canister code

### PR DESCRIPTION
**Pull request by bogwar**
This PR updates: https://sdk.dfinity.org/docs/developers-guide/concepts/canisters-code.html

### CODE REVIEW

File: modules/developers-guide/pages/concepts/canisters-code.adoc [hunk 1]
```diff
@@ -4,27 +4,27 @@
 :IC: Internet Computer
 :company-id: DFINITY
 
-One of the most important principles to keep in mind is that the {IC} is primarily a distributed and decentralized platform for running software. 
+One of the most important principles to keep in mind is that the {IC} is primarily a blockchain that allows running software in a distributed, replicated way.
```
Comment (jancamenisch) [inline modules/developers-guide/pages/concepts/canisters-code.adoc:7]: remove "primarily"

File: modules/developers-guide/pages/concepts/canisters-code.adoc [hunk 2]
```diff
@@ -4,27 +4,27 @@
...
+A canister smart contract is similar to a container in that both are deployed as a software unit that contains compiled code and dependencies for an application or service.
 
 Containerization allows for applications to be decoupled from the environment, allowing for easy and reliable deployment.
-The canister differs from a container, however, in that it also stores information about the current software *state* with a record of preceding events and user interactions. 
+The canister differs from a container, however, in that it also stores information about the current software *state*.
 
-While a containerized application might include information about the state of the environment in which the application runs, a software canister is able to persist a record of state changes that resulted from an application’s functions being used.
+While a containerized application might include information about the state of the environment in which the application runs, a canister smart contract is able to persist a record of state changes that resulted from its functions being called.
```
Comment (jancamenisch) [inline modules/developers-guide/pages/concepts/canisters-code.adoc:22]: can delete smart contract here.

File: modules/developers-guide/pages/concepts/canisters-code.adoc [hunk 3]
```diff
@@ -34,30 +34,30 @@ This concept of a canister consisting of both program and state is an important
...
+There is an inherent tradeoff between security and performance: the result is trustworthy because two-thirds of the replicas in a subnet must agree on the result, but the call is slow.
 * The called canister can invoke functions exposed by other canisters
 |===
 
 As a developer, it is important to recognize this relationship between the calls that query the canister and the calls that change the canister state. 
 In particular, you should keep in mind the inherent tradeoff between security and performance.
 
 [[dev-motoko-intro]]
-== How to develop applications for the {IC}
+== How to develop descentralized applications (dapps) for the {IC}
```
Comment (jancamenisch) [inline modules/developers-guide/pages/concepts/canisters-code.adoc:56]: just say dapp

File: modules/developers-guide/pages/concepts/canisters-code.adoc [hunk 4]
```diff
@@ -34,30 +34,30 @@ This concept of a canister consisting of both program and state is an important
...
 
 | Update calls
 a| Allow the user to change the state of the canister and have *changes persisted*.
 
 * Are answered asynchronously.
 * Must pass through consensus to return the result. 
-Because consensus is required, changing the state of a canister can take time. Therefore, update calls use the actor-based programming model (with state isolation) to allow concurrent and asynchronous processing. 
-There is an inherent tradeoff between security and performance because two-thirds of the replicas in a subnet must agree on the result.
+Because consensus is required, changing the state of a canister, and returning the result can take time.
+There is an inherent tradeoff between security and performance: the result is trustworthy because two-thirds of the replicas in a subnet must agree on the result, but the call is slow.
```
Comment (jancamenisch) [inline modules/developers-guide/pages/concepts/canisters-code.adoc:48]: Should we refer to certified variables here?

**Review by jancamenisch — CHANGES_REQUESTED** [review]

File: modules/developers-guide/pages/concepts/canisters-code.adoc (hunk 4 above)
Comment (bogwar) [reply modules/developers-guide/pages/concepts/canisters-code.adoc:48]: Not really: the emphasis is on the difference between update and query calls and mentioning certified variables here I think would be distracting from the main message

**Review by jancamenisch — APPROVED** [review]

**Pull request merged**